### PR TITLE
initial mypy support

### PR DESCRIPTION
**Pull request by autoferrit**
For now, just silencing a few items im not sure how to type properly yet, and doing some basic typings in actions.py. also adding it so it can be run in travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ install:
   - pipenv install --deploy --dev --ignore-pipfile
 
 script:
+  - pipenv run mypy slack/
```
Comment (ovv) [inline .travis.yml]: Could you add that into the `slack/tests/test.sh` script instead

File: slack/actions.py [hunk 2]
```diff
@@ -85,7 +86,7 @@ def register(self, callback_id, handler, name="*"):
 
         self._routes[callback_id][name].append(handler)
 
-    def dispatch(self, action):
+    def dispatch(self, action: Dict[str, Any]):
```
Comment (ovv) [inline slack/actions.py]: action should be of type `Action` of the same file (same for the other method :point_down: )

File: slack/tests/conftest.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ def _default_response(response):
     return response
 
 
-@pytest.fixture(params={**data.Events.__members__, **data.Messages.__members__})
+@pytest.fixture(params={**data.Events.__members__, **data.Messages.__members__}) # type: ignore
```
Comment (ovv) [inline slack/tests/conftest.py]: I guess these are some weird cases ? If you have any idea for improving all that feel free to share :)

**Review by ovv — CHANGES_REQUESTED** [review]
Looks pretty straightforward :+1:

File: slack/tests/conftest.py (hunk 3 above)
Comment (autoferrit) [reply slack/tests/conftest.py]: Yea these im not sure yet. but im just silencing them for now until i do some more typing and get a better idea when I should put here if anything. When I get home, ill paste the error mypy is showing to see if you have an idea.

**Comment by autoferrit** [comment]
Last build on 3.6 simply errored out. last fix im hoping fixed the issue. can it just be restarted?

**Comment by ovv** [comment]
restarted but it fails on documentation build.
If it's blocking I might skip that for 3.6 since readthedoc build it with 3.7 anyways

**Comment by ovv** [comment]
It should work if you rebase on master. I disabled check for non 3.7 version

**Comment by autoferrit** [comment]
Woohoo. works now. I squashed the commits as well when I was rebasing.

**Review by ovv — APPROVED** [review]

**Pull request merged**